chart/kubeapps/templates/kubeappsapis/rbac.yaml

```diff
@@ -24,6 +24,7 @@ rules:
       - "source.toolkit.fluxcd.io"
     resources: ['*']
     verbs: ['*']
+{{- if .Values.kubeappsapis.unsafeUseDemoSA }}
```

Comment:
Exactly, that was my idea. I don't really know how the RBAC of flux/kapp is being set by default, so I just wanted to have the ClusterRole in case if we need to create the binding for ourselves.

Once we get rid of the `unsafeUseDemoSA`, I guess we can remove the whole RBAC, as it is not something that Kubeapps should care about (as you mentioned).